src/main/kotlin/org/team5419/fault/hardware/PS4Controller.kt

```diff
@@ -3,73 +3,134 @@ package org.team5419.fault.hardware
-        return getBButtonReleased()
+    public fun getTrigger(hand: Hand): Boolean {
+        when (hand) {
+            Hand.kLeft -> return getRawButton(LEFT_TRIGGER_PORT)
+            Hand.kRight -> return getRawButton(RIGHT_TRIGGER_PORT)
+        }
     }
 
-    public fun getAButton(): Boolean {
-        return getBButton()
-    }
+    // Share Button
+    public fun getShareButtonPressed(): Boolean { return getRawButtonPressed(SHARE_BUTTON_PORT) }
+    public fun getShareButtonReleased(): Boolean { return getRawButtonReleased(SHARE_BUTTON_PORT) }
+    public fun getShareButton(): Boolean { return getRawButton(SHARE_BUTTON_PORT) }
 
-    public fun getBButtonPressed(): Boolean {
-        return getXButtonPressed()
-    }
+    // Options Button
+    public fun getOptionsButtonPressed(): Boolean { return getRawButtonPressed(OPTIONS_BUTTON_PORT) }
+    public fun getOptionsButtonReleased(): Boolean { return getRawButtonReleased(OPTIONS_BUTTON_PORT) }
+    public fun getOptionsButton(): Boolean { return getRawButton(OPTIONS_BUTTON_PORT) }
 
-    public fun getBButtonReleased(): Boolean {
-        return getXButtonReleased()
+    // Left & Right Sticks
+    public fun getStickButtonPressed(hand: Hand): Boolean {
+        when (hand) {
+            Hand.kLeft -> return getRawButtonPressed(LEFT_STICK_BUTTON_PORT)
+            Hand.kRight -> return getRawButtonPressed(RIGHT_STICK_BUTTON_PORT)
+        }
     }
 
-    public fun getBButton(): Boolean {
-        return getXButton()
+    public fun getStickButtonReleased(hand: Hand): Boolean {
+        when (hand) {
+            Hand.kLeft -> return getRawButtonReleased(LEFT_STICK_BUTTON_PORT)
+            Hand.kRight -> return getRawButtonReleased(RIGHT_STICK_BUTTON_PORT)
+        }
     }
 
-    public fun getXButtonPressed(): Boolean {
-        return getAButtonPressed()
+    public fun getStickButton(hand: Hand): Boolean {
+        when (hand) {
+            Hand.kLeft -> return getRawButton(LEFT_STICK_BUTTON_PORT)
+            Hand.kRight -> return getRawButton(RIGHT_STICK_BUTTON_PORT)
+        }
     }
 
-    public fun getXButtonReleased(): Boolean {
-        return getAButtonReleased()
-    }
+    // Home Button
+    public fun getHomeButtonPressed(): Boolean { return getRawButtonPressed(HOME_BUTTON_PORT) }
+    public fun getHomeButtonReleased(): Boolean { return getRawButtonReleased(HOME_BUTTON_PORT) }
+    public fun getHomeButton(): Boolean { return getRawButton(HOME_BUTTON_PORT) }
 
-    public fun getXButton(): Boolean {
-        return getAButton()
-    }
+    // Touch Pad
+    public fun getTouchPadPressed(): Boolean { return getRawButtonPressed(TOUCHPAD_BUTTON_PORT) }
+    public fun getTouchPadReleased(): Boolean { return getRawButtonReleased(TOUCHPAD_BUTTON_PORT) }
+    public fun getTouchPad(): Boolean { return getRawButton(TOUCHPAD_BUTTON_PORT) }
+
+    // POV
+    public fun getPOVUp(error: Int): Boolean { return getPOV() <= 0 + error && getPOV() >= 360 - error }
```

Comment:
Make this `getPOV() <=1` instead of `getPOV() <= 0`. This is because when the controller is unplugged, `getPOV()` will return `0`. That means that if you call getPOVUp() when the controller is unplugged, it will return true. Which is bad!